community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/pipes/ExpandIntoPipe.scala

```diff
@@ -35,33 +43,21 @@ case class ExpandIntoPipe(source: Pipe,
                           types: LazyTypes)(val estimatedCardinality: Option[Double] = None)
                          (implicit pipeMonitor: PipeMonitor)
   extends PipeWithSource(source, pipeMonitor) with RonjaPipe {
-
   self =>
 
+  private val CACHE_SIZE = 100000
+  //cache of known connected nodes
+  private val connected = new LRUCache[(Node, Node), Seq[Relationship]](CACHE_SIZE)
```

Comment:
Doesn't declaring this as a field here tie the life cycle of this cache to the life of the query plan.
It would be fine to "cache" things while the query is executing, but caching things across runs will produce bad results.